TrackingPerformance/Condor/condorJobs_sim.py

```diff
@@ -1,156 +1,248 @@
+    except FileExistsError:
+        print(
+            f"Error: Directory '{directory_jobs}' already exists and should not be overwritten."
         )
-        bash_file = directory_jobs + f"/bash_script_{dect}_{part}_{momentum}_{theta}_{task_index}.sh"
-        with open(bash_file, "w") as file:
-            file.write(bash_script)
-            file.close()
-
-if not need_to_create_scripts:
-    print("All output files are correct.")
-    sys.exit(0)
-
-# ============================
-# Condor Submission Script
-# ============================
-# Write the condor submission script
-condor_script = (
-    "executable = $(filename) \n"
-    "arguments = $(ClusterId) $(ProcId) \n"
-    "output = output.$(ClusterId).$(ProcId).out \n"
-    "error = error.$(ClusterId).$(ProcId).err \n"
-    "log = log.$(ClusterId).log \n"
-    f"+JobFlavour = \"{JobFlavour}\" \n"  
-    "queue filename matching files *.sh \n"
-)
-condor_file = directory_jobs + "/condor_script.sub"
-with open(condor_file, "w") as file2:
-    file2.write(condor_script)
-    file2.close()
-
-# ====================
-# Submit Job to Condor
-# ====================
-os.system("cd "+ directory_jobs + "; condor_submit condor_script.sub")
+        sys.exit(1)
+
+    SIMEOSDir.mkdir(
+        parents=True, exist_ok=True
+    )  # This will create the directory if it doesn't exist, without raising an error if it does
+
+    # =======================
+    # Simulation Job Creation
+    # =======================
+
+    # Create all possible combinations
+    import itertools
+
+    iter_of_combined_variables = itertools.product(
+        thetaList_, momentumList_, particleList_, DetectorModelList_
+    )
+
+    need_to_create_scripts = False
+
+    for theta, momentum, part, dect in iter_of_combined_variables:
+        for task_index in range(N_jobs_per_para_set):
+
+            output_file_name_parts = [
+                f"SIM_{dect}",
+                f"{part}",
+                f"{theta}_deg",
+                f"{momentum}_GeV",
+                f"{Nevt_per_job}_evts",
+                f"{task_index}",
+            ]
+            output_file_name = Path("_".join(output_file_name_parts)).with_suffix(
+                ".edm4hep.root"
+            )
 
+            # Check if the output file already exists and has correct Nb of events
+            output_dir = SIMEOSDir / part
+            output_dir.mkdir(parents=True, exist_ok=True)
+            output_file_path = output_dir / output_file_name
+            if CHECK_OUTPUT and output_file_path.exists():
+                root_file = ROOT.TFile(output_file_path, "READ")
+                events_tree = root_file.Get("events")
+                if events_tree:  # FIXME: why no else?
+                    if events_tree.GetEntries() == int(Nevt_per_job):
+                        root_file.Close()
+                        continue
+                root_file.Close()
+            else:
+                need_to_create_scripts = True
```

Comment:
This check looks like we only rewrite the files when we change the number of events per job? So if we just change the type of particles they are not written?

Reply:
From what I can see, there doesn't seem to be much change compared to what I did. This check will rerun the simulation for the missing files because the jobs crashed or encountered other issues, which happens quite often. Therefore, you need to use the same variables. It will also verify if the output files contain the correct number of events. I've encountered this issue several times where the files exist but contain only a few events.

Reply:
To me, it seems like the `need_to_create_scripts` variable is only set to `True` if the condition in line `161` equals `False`. The if statement `events_tree.GetEntries() == int(Nevt_per_job)` misses an `else` to ensure that missing events are produced. I will create an issue based on it to keep this pull request cleaner